haystack/agents/conversational.py

```diff
@@ -1,63 +1,87 @@
 
     Here is an example of how you can create a simple chat application:
     ```
     import os
 
-    from haystack.agents.base import ConversationalAgent
-    from haystack.agents.memory import ConversationSummaryMemory
+    from haystack.agents.conversational import ConversationalAgent
     from haystack.nodes import PromptNode
+    from haystack.tools import ToolsManager, Tool
 
-    pn = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
-    agent = ConversationalAgent(pn, memory=ConversationSummaryMemory(pn))
+    # Initialize a PromptNode and a ToolsManager with the desired tools
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+    tools_manager = ToolsManager([Tool(name="ExampleTool", pipeline_or_node=example_tool_node)])
 
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node, tools_manager=tools_manager)
+
+    # Use the agent in a chat application
     while True:
         user_input = input("Human (type 'exit' or 'quit' to quit): ")
         if user_input.lower() == "exit" or user_input.lower() == "quit":
             break
-        elif user_input.lower() == "memory":
-            print("\nMemory:\n", agent.memory.load())
         else:
             assistant_response = agent.run(user_input)
             print("\nAssistant:", assistant_response)
-
     ```
     """
 
     def __init__(
         self,
         prompt_node: PromptNode,
         prompt_template: Optional[Union[str, PromptTemplate]] = None,
-        memory: Optional[Memory] = None,
-        prompt_parameters_resolver: Optional[Callable] = None,
+        tools_manager: Optional[ToolsManager] = None,
+        memory: Optional[Memory] = ConversationMemory(),
+        prompt_parameters_resolver: Optional[Callable] = react_parameter_resolver,
+        max_steps: int = 5,
+        final_answer_pattern: str = r"Final Answer\s*:\s*(.*)",
+        streaming: bool = True,
     ):
         """
         Creates a new ConversationalAgent instance
 
         :param prompt_node: A PromptNode used to communicate with LLM.
         :param prompt_template: A string or PromptTemplate instance to use as the prompt template. If no prompt_template
         is provided, the agent will use the "conversational-agent" template.
-        :param memory: A memory instance for storing conversation history and other relevant data, defaults to
-        ConversationMemory.
+        :param tools_manager: A ToolsManager instance to manage tools for the agent. If no tools_manager is provided,
+        the agent will install an empty ToolsManager instance.
+        :param max_steps: The maximum number of steps for the agent to take, defaults to 5.
+        :param memory: A memory object for storing conversation history and other relevant data, defaults to
+        ConversationSummaryMemory if no memory is provided.
```

Comment:
AFAICS the default is `ConversationMemory` instead of `ConversationSummaryMemory`.

```suggestion
        :param memory: A memory object for storing conversation history and other relevant data, defaults to
        ConversationMemory if no memory is provided.
```